Fix potential race condition in builds leading to errors like:
fatal error: mlir/Dialect/Func/IR/FuncOps.h.inc: No such file or directory

From ef65095cdb845f401eb64ff8eedb6efcd118e46d Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Sat, 6 Jul 2024 11:57:54 +0200
Subject: [PATCH] [mlir] Fix missing FuncOps.h.inc

Closes #84568
---
 mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt b/mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt
index 61e49b0da8b2d..3a1996349dbed 100644
--- a/mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt
+++ b/mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt
@@ -8,6 +8,9 @@ add_mlir_dialect_library(MLIRAffineAnalysis
   ADDITIONAL_HEADER_DIRS
   ${MLIR_MAIN_INCLUDE_DIR}/mlir/Dialect/Affine
 
+  DEPENDS
+  MLIRFuncOpsIncGen
+
   LINK_LIBS PUBLIC
   MLIRAffineDialect
   MLIRAnalysis
